src/frontend/src/handler/create_source.rs

```diff
@@ -750,6 +750,18 @@ pub(crate) fn bind_all_columns(
     }
 }
 
+fn hint_upsert(encode: &Encode) -> String {
+    format!(
+        r#"Hint: For FORMAT UPSERT ENCODE {encode:}, INCLUDE KEY must be specified and the key column must be used as primary key.
```

Comment:
How I wish `thiserror` provides something similar to this:

https://docs.rs/snafu/latest/snafu/derive.Snafu.html#providing-data-beyond-the-error-trait